### PR TITLE
meta/sql: fix wrong result from scanning SQL database

### PR DESCRIPTION
**Pull request by davies**
There is a bug in xorm when scanning a binary column, it could return partial or wrong result.

This PR fix that by reset the binary column before scan next row.

### CODE REVIEW

**Comment by davies** [comment]
Create a bug for xorm: https://gitea.com/xorm/xorm/issues/2134

File: pkg/meta/sql.go [hunk 1]
```diff
@@ -1622,6 +1631,7 @@ func (m *dbMeta) doFindStaleSessions(limit int) ([]uint64, error) {
 	}
 	var sids []uint64
 	for rows.Next() {
+		s.Info = nil
```
Comment (SandyXSD) [inline pkg/meta/sql.go:1634]: no need since s.Info is not used

File: pkg/meta/sql.go [hunk 2]
```diff
@@ -764,7 +773,7 @@ func (m *dbMeta) Truncate(ctx Context, inode Ino, flags uint8, length uint64, at
 				return err
 			}
 			for rows.Next() {
-				if err = rows.Scan(&c); err != nil {
+				if err = scanChunk(rows, &c); err != nil {
```
Comment (SandyXSD) [inline pkg/meta/sql.go:776]: no need since Slices is not used

File: pkg/meta/sql.go [hunk 3]
```diff
@@ -1976,7 +1986,7 @@ func (m *dbMeta) doDeleteFileData(inode Ino, length uint64) {
 	}
 	var indexes []uint32
 	for rows.Next() {
-		if rows.Scan(&c) == nil {
+		if scanChunk(rows, &c) == nil {
```
Comment (SandyXSD) [inline pkg/meta/sql.go:1989]: no need

File: pkg/meta/sql.go [hunk 4]
```diff
@@ -2188,6 +2198,7 @@ func (m *dbMeta) ListXattr(ctx Context, inode Ino, names *[]byte) syscall.Errno
 	defer rows.Close()
 	*names = nil
 	for rows.Next() {
+		x.Value = nil
```
Comment (SandyXSD) [inline pkg/meta/sql.go:2201]: no need

File: pkg/meta/sql.go (hunk 2 above)
Comment (davies) [reply pkg/meta/sql.go:776]: It's better to use the same way

**Pull request merged**